docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -953,12 +953,18 @@ Cython code.  Here is the list of currently supported directives:
     Copy the original source code line by line into C code comments in the generated
     code file to help with understanding the output.
     This is also required for coverage analysis.
-    
+
 ``cpp_locals`` (True / False)
     Make C++ variables behave more like Python variables by allowing them to be
     "unbound" instead of always default-constructing them at the start of a
     function.  See :ref:`cpp_locals directive` for more detail.
 
+``legacy_implicit_noexcept`` (True / False)
```

Comment:
I am not native english speaker so I can be cumbersome. I would like to somehow discourage users using this directive at all. Can we mention somehow that this directive is just for migration. I was thinking about mentioning that this directive is temporary for migration purposes and will be removed in the future?

Reply:
I think:

> This directive is intended to help migrate legacy code written before Cython 3. We recommend leaving it off for new code.

I don't think there's a need to threaten to remove it yet.

Reply:
thanks!

Reply:
I only saw this part of the discussion now. I think it's ok to tell users that it will eventually be removed. That's what "deprecation" means, really. We may find a wording that keeps them from perpetually asking "are we there yet", though.

Reply:
Commited @scoder suggestion with message of future removal. Marking as resolved.